[Zhao Yiheng] iP

linedoestrolling opened the pull request:
# Elysia
> Good morning. A new day should start with a lovely encounter~


Elysia frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/linedoestrolling/ip).

2. double-click it.

3. add your tasks.

4. let it manage your tasks for you 😉

And it is **FREE**!


Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Reminders

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

linedoestrolling committed:
Change bot name to raiden and implement skeleton

linedoestrolling committed:
Implement echo and add customisation

linedoestrolling committed:
Change bot to Elysia and add list

linedoestrolling committed:
Implement mark as done

linedoestrolling committed:
Implement todo, event and deadline

Only one type of event.

The aforementioned event types are classes extending from event as
the description and to string behaviour are similar.

linedoestrolling committed:
Semi-automate testing of Elysia

All commands are run at least once to test output

linedoestrolling committed:
Add error handling

Basic errors for cases like wrong command, empty arguments and
marking events that do not exist handled.

linedoestrolling committed:
Implement delete

Changed TaskList to use arraylist and added test cases in
automated testing

linedoestrolling committed:
Add saving and reading to data.txt

linedoestrolling committed:
Merge branch-Level-7 and master

linedoestrolling committed:
Add date to deadline

linedoestrolling committed:
Merge branch-Level-8 and master

linedoestrolling committed:
Refactor into packages

linedoestrolling committed:
Merge add-gradle-support and master

publicmain commented on src/main/java/TaskList.java:
```diff
@@ -1,37 +1,41 @@
 public class TaskList {
-    private Task[] list;
-    private int listPointer = 0;
+    private ArrayList<Task> list;
```
I like the shift from an array-based implementation to using `ArrayList`. This will likely make the code more flexible and reduce the risk of `ArrayIndexOutOfBoundsException`. Was this change made in response to a specific limitation you encountered with the array-based approach?

publicmain commented on src/main/java/TaskList.java:
```diff
@@ -1,37 +1,41 @@
+        list.add(task);
+    }
+
+    public Task deleteTask(int index) {
```
The addition of the `deleteTask` functionality is a great enhancement. I noticed that you're checking if the input equals `"delete"` for handling the absence of an index. Have you considered providing more user-friendly feedback when an index is not provided? For example, a message prompting the user to specify which task to delete might help improve the user experience.

publicmain reviewed (commented):
I noticed you've improved the task handling logic significantly. One suggestion for future enhancement could be to implement a way to list all tasks when a user inputs an invalid command, offering them a visual guide to the tasks they can manage. What do you think?

publicmain commented on src/main/java/FileReaderWriter.java:
```diff
@@ -2,6 +2,7 @@
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.util.Objects;
+import java.util.Scanner;
```
I noticed that you've switched from using `BufferedReader` to `Scanner`, which makes the code more concise and the file reading logic clearer. I like this improvement! Did you choose `Scanner` because of specific features that better fit the current needs, or was it mainly to reduce the complexity of the code?

publicmain commented on src/main/java/FileReaderWriter.java:
```diff
@@ -38,47 +39,38 @@ public String writeFile() {
-                break;
             }
-        }
+            output = "I loaded your past data! Aren't I amazing? Come on praise me!";
```
The friendly feedback message "I loaded your past data! Aren't I amazing? Come on praise me!" after successfully loading data is a nice touch! It adds a fun and engaging element to the user experience. I really like this approach.

publicmain reviewed (commented):
Overall, I appreciate the thoughtful improvements you've made in this code. The changes enhance both the functionality and readability, making it more maintainable for the future. Great work! Looking forward to seeing how this evolves further.

calebchongsj commented on src/main/java/elysia/ui/FileReaderWriter.java:
```diff
@@ -0,0 +1,83 @@
+                String taskInput = sc.nextLine();
+                String[] taskParts = taskInput.split("\\|");
+                switch(taskParts[0]) {
+                case "T":
```
Great to see that you have indented your case statements inside the switch block.

calebchongsj commented on src/main/java/elysia/ui/InputOutputHandler.java:
```diff
@@ -0,0 +1,114 @@
+    TaskList taskList;
+    FileReaderWriter fileReaderWriter;
+
+    public InputOutputHandler() {
```
Is this class equivalent to the Parser class?

calebchongsj commented on src/main/java/elysia/ui/InputOutputHandler.java:
```diff
@@ -0,0 +1,114 @@
+                }
+            }
+        }
+        else {
```
Indent the else statement on the same line as the closing braces of the else if block.

calebchongsj commented on src/main/java/elysia/ui/InputOutputHandler.java:
```diff
@@ -0,0 +1,114 @@
+    public InputOutputHandler() {
+        taskList = new TaskList();
+        fileReaderWriter = new FileReaderWriter(taskList);
+        String msg = fileReaderWriter.readFile();
```
Is this message referring to the text from the file?

calebchongsj commented on src/main/java/elysia/ui/InputOutputHandler.java:
```diff
@@ -0,0 +1,114 @@
+        }
+    }
+
+    public boolean parseInput(String input) throws ElysiaException, StringIndexOutOfBoundsException {
```
Input name should be more specific.

calebchongsj reviewed (commented):
Great job so far. Just some nits to fix in the codebase. Keep up the good work.

linedoestrolling committed:
Setup gradle and add Junit tests

linedoestrolling committed:
Add JavaDocs for all Java files

linedoestrolling committed:
Fix some minor coding standard issues

linedoestrolling committed:
Add find tasks by keyword

linedoestrolling committed:
Merge branch 'branch-A-JavaDoc'

linedoestrolling committed:
Fix merge conflicts

linedoestrolling committed:
Merge branch 'branch-Level-9'

linedoestrolling committed:
Add GUI and do major refactoring

InputOutputHandler handles both parsing and executing commands.

The parse method is too long and confusing.

Extract different commands into their own classes which extend
an abstract Command class. Add more specific exceptions for
different commands.

This will enhance code readability and make future work easier.

linedoestrolling committed:
Merge branch-Level-10 and master

No GUI, only text based

GUI will enhance user experience

Use JavaFX to build simple GUI

Javafx provides a simple but good looking GUI

linedoestrolling committed:
Add assertions to some commands

Commands do not have assertions

Add assertions about tasklist length

This will ensure minimal bugs in commands

linedoestrolling committed:
Do minor refactoring

No comments for commands and exceptions

Important for code readability

Add comments and remove unused packages

linedoestrolling committed:
Merge pull request #2 from linedoestrolling/branch-A-Assertions

Add assertions to some commands

linedoestrolling committed:
Fix merge conflicts

linedoestrolling committed:
Merge pull request #3 from linedoestrolling/branch-A-CodeQuality

Do minor refactoring

linedoestrolling committed:
Add support for recurring tasks

Currently no support for recurring tasks.

User has to manually add back recurring tasks.

Recurring tasks will now not mark itself when marked.
The same mark message will be shown but the task will remain
unmarked.

This will enhance user experience. In the future, support for
different dates for recurring tasks could be added.

linedoestrolling committed:
Fix styles

linedoestrolling committed:
Update README.md

linedoestrolling committed:
Fix no data.txt bug

linedoestrolling committed:
Fix bye not working properly

linedoestrolling committed:
Update README.md with resources

linedoestrolling committed:
Fix README.md formatting

linedoestrolling committed:
Move README.md to /docs/

linedoestrolling committed:
Update README in docs
